scripts/generate-sitemap.js

```diff
@@ -1,78 +1,157 @@
-const fs = require("fs").promises;
-const path = require("path");
+const fs = require("fs");
+const fsp = fs.promises;
+const _path = require("path");
 
-const schemaDirPath = path.join(__dirname, "../schema");
-const sitemapPath = path.join(__dirname, "../sitemap.json");
+const schemaDirPath = _path.join(__dirname, "../schema");
+const sitemapPath = _path.join(__dirname, "../sitemap.json");
 
 async function readSchemaJson(filePath) {
   try {
-    const schemaContent = await fs.readFile(filePath, "utf8");
+    const schemaContent = await fsp.readFile(filePath, "utf8");
     return JSON.parse(schemaContent);
   } catch (error) {
     return null;
   }
 }
 
-async function processDirectory(dirPath, relativePath = "") {
-  const entries = await fs.readdir(dirPath, { withFileTypes: true });
-  let paths = [];
-
-  for (const entry of entries) {
-    const entryPath = path.join(dirPath, entry.name);
+async function processFile(fullPath, relativePath, name) {
+  const permalink = relativePath.split(".")[0];
+  const schemaData = await readSchemaJson(fullPath);
+  if (!schemaData) {
+    return null;
+  }
+  const pageName = name.split(".")[0].replace(/-/g, " ");
+  const siteMapEntry = {
+    permalink,
+    title:
+      schemaData.page.title ||
+      pageName.charAt(0).toUpperCase() + pageName.slice(1),
+  };
 
-    if (entry.name === "index.json") {
-      continue;
+  // check if file is actually an index page for a directory
+  const directoryPath = fullPath.split(".")[0];
+  if (fs.existsSync(directoryPath)) {
+    const isDirectory = (await fsp.stat(directoryPath)).isDirectory();
+    if (isDirectory) {
+      return {
+        ...siteMapEntry,
+        children: await processDirectory(directoryPath, permalink),
+      };
     }
+  }
+  return siteMapEntry;
+}
+
+// generates sitemap entries and an index file for directories without an index file
+async function processDanglingDirectory(fullPath, relativePath, name) {
+  const children = await processDirectory(fullPath, relativePath);
+  // TODO: Improve the content for generated index pages
+  const listOfChildPages = {
+    type: "unorderedlist",
+    items: children.map(
+      (child) => `<a href=${child.permalink}>${child.title}</a>`
```

Comment:
nit: not sure if we need to add quotes here

```suggestion
      (child) => `<a href="${child.permalink}">${child.title}</a>`
```

Reply:
seems fine
<img width="995" alt="Screenshot 2024-04-05 at 1 47 38 PM" src="https://github.com/isomerpages/isomer-next-build/assets/77665182/914acef3-8ca2-46fc-87b7-ddeaa1c0b884">